pkg/apis/servicecatalog/types.go

```diff
@@ -270,9 +270,17 @@ type InstanceSpec struct {
 	// provisioned from.
 	PlanName string
 
-	// Parameters is a YAML representation of the properties to be
+	// List of sources to populate parameters.
+	// When a key exists in multiple
+	// sources, the value associated with the last source will take precedence.
+	// Values defined by a Parameter with a duplicate key will take precedence.
+	// +optional
+	ParametersFrom []ParametersFromSource
+
+	// Parameters is a set of the parameters to be
 	// passed to the underlying broker.
-	Parameters *runtime.RawExtension
+	// +optional
+	Parameters []Parameter
```

Comment:
> So for now we're dropping parameters and you can only set things from parametersFrom field where the value is the full json (either inlined or in a secret) ?

Exactly, except we will deprecate the current parameters field so users have notice that it will be removed.

All of the other API constructions we've discussed are backward compatible from what I described, so we can add behaviors in the future as desired, and solve the most immediate problems now.